webapp/channels/src/components/signup/signup.tsx

```diff
@@ -368,6 +369,19 @@ const Signup = ({onCustomizeHeader}: SignupProps) => {
         }
     }, [onCustomizeHeader, handleHeaderBackButtonOnClick, isMobileView, getAlternateLink, search]);
 
+    useEffect(() => {
```

Comment:
Instead of this complicated `useEffect`, what about focusing the invalid input directly in `handleSubmit`? It's a bit less reactive, but that logic feels a bit easier to follow to me

Reply:
It's actually messier IMO to have to put a bunch of checks and stops in `isUserValid`, since we still want to focus the first errored input while displaying errors for the rest.

I think this is what `useEffect` does so well, even if it does make you look in multiple places for code

Reply:
I do feel like there's a bit of a disconnect between the validation logic and the `useEffect`, and I worry about the `useEffect` triggering accidentally if it relies on some other prop somehow, but your point about handling multiple invalid inputs makes sense. I'm trying to think if there's a better pattern we could use that would satisfy both of those, but nothing notably better is coming to mind right now